src/designer/elsa-workflows-studio/src/components/screens/workflow-definition-editor/elsa-workflow-definition-editor-screen/elsa-workflow-definition-editor-screen.tsx

```diff
@@ -381,19 +411,112 @@ export class ElsaWorkflowDefinitionEditorScreen {
     this.handleConnectionContextMenuChange({x: 0, y: 0, shown: false, activity: null});
   }
 
-  onActivityContextMenuButtonClicked(e: CustomEvent<ActivityContextMenuState>) {
+  async onActivityContextMenuButtonClicked(e: CustomEvent<ActivityContextMenuState>) {
     this.activityContextMenuState = e.detail;
   }
 
+  async onActivityContextMenuButtonTestClicked(e: CustomEvent<ActivityContextMenuState>) {
+    this.activityContextMenuTestState = e.detail;
+    this.selectedActivityId = e.detail.activity.activityId;
+
+    if (!e.detail.shown) {
+      return;
+    }
+  }  
+
+  async onActivitySelected(e: CustomEvent<ActivityModel>) {
+    this.selectedActivityId = e.detail.activityId;
+  }
+
+  async onActivityDeselected(e: CustomEvent<ActivityModel>) {
+    if (this.selectedActivityId == e.detail.activityId)
+      this.selectedActivityId = null;
+  }
+  
   onConnectionContextMenuButtonClicked(e: CustomEvent<ActivityContextMenuState>) {
     this.connectionContextMenuState = e.detail;
   }
 
+  onTestActivityMessageReceived = async args => {
+    const message = args as WorkflowTestActivityMessage;
+    if (!!message) {
+      this.workflowTestActivityMessages = this.workflowTestActivityMessages.filter(x => x.activityId !== message.activityId);
+      this.workflowTestActivityMessages = [...this.workflowTestActivityMessages, message];
+    }
+    else
+      this.workflowTestActivityMessages = [];
+
+    this.render();
+  };
+
+  async updateHttpEndpointJsonSchema(request: WorkflowTestUpdateHttpEndpoint) {
+    let workflowModel = {...this.workflowModel};
+    const activity = workflowModel.activities.find(x => x.activityId == request.activityId);
+    if (activity)
+    {
+      const activityModel = activity.properties.find(x => x.name == 'Schema');
+      activityModel.expressions[SyntaxNames.Literal] = request.jsonSchema;
+      eventBus.emit(EventTypes.UpdateActivity, this, activityModel);
+    }
+  };    
```

Comment:
This code is specific to the HTTP Endpoint activity, so you should find a way to decouple the workflow definition editor from this logic that is specific to the HTTP Endpoint by leveraging the HTTP Endpoint plugin.